extensions-core/orc-extensions/src/test/java/org/apache/druid/data/input/orc/OrcReaderTest.java

```diff
@@ -314,8 +314,8 @@ public void testJsonPathFunctions() throws IOException
         //deviation of [7,8,9] is 1/3, stddev is sqrt(1/3), approximately 0.8165
         Assert.assertEquals(0.8165, Double.parseDouble(Iterables.getOnlyElement(row.getDimension("stddev"))), 0.0001);
 
-        //append is not supported
-        Assert.assertEquals(Collections.emptyList(), row.getDimension("append"));
+        //Support for append has been added in json-path-2.9.0
+        //Assert.assertEquals(Collections.emptyList(), row.getDimension("append"));
```

Comment:
if append is now supported, then let's update the test to check that it does what it needs to do.

Reply:
` row.getDimension("append")` results in `java.lang.UnsupportedOperationException: Unused` coming from https://github.com/apache/druid/blob/0089f6b905f10a498506a747f0e05275fe366851/processing/src/main/java/org/apache/druid/java/util/common/parsers/FlattenerJsonProvider.java#L118 as we have overriden the parse function.

Reply:
Therefore, while the upstream has added the capability of `append` , we are not using it.

Reply:
understood, but there are two issues here:
1. as a principle we don't leave commented code unless there's a very good reason.
2. This is potentially being breaking change. Any json-path expression using append today would have silently been ignored, but would now cause an error.

My suggestion would be to:
1. update the test to check that we do get the exception
2. make a note in the PR description that we should call this out in the release notes as a potential issue to watch out for when upgrading.

Reply:
Updated